src/status_im/subs/wallet/activities.cljs

```diff
@@ -1,26 +1,82 @@
+           :amount       (activity-amount amount-out)
+           :network-name network-name
+           :network-logo (quo.resources/get-network network-name))))
+
+(defn- process-receive-activity
+  [{:keys [symbol-in amount-in chain-id-in]} activity chain-id->network-name]
+  (let [network-name (chain-id->network-name chain-id-in)]
+    (assoc activity
+           :transaction  :receive
+           :token        symbol-in
+           :amount       (activity-amount amount-in)
+           :network-name network-name
+           :network-logo (quo.resources/get-network network-name))))
+
+(defn- process-activity-by-type
+  [chain-id->network-name
+   {:keys [activity-type activity-status timestamp sender recipient] :as data}]
+  (let [activity {:relative-date (datetime/timestamp->relative (* timestamp 1000))
+                  :timestamp     timestamp
+                  :status        (constants/wallet-activity-status->name activity-status)
+                  :sender        sender
+                  :recipient     recipient}]
+    (condp = activity-type
+      constants/wallet-activity-type-send
+      (process-send-activity data activity chain-id->network-name)
+
+      constants/wallet-activity-type-receive
+      (process-receive-activity data activity chain-id->network-name)
+
+      nil)))
+
+(rf/reg-sub
+ :wallet/activities-for-current-viewing-account
  :<- [:wallet/all-activities]
  :<- [:wallet/current-viewing-account-address]
- (fn [[activities current-viewing-account-address]]
-   (->> activities
-        (filter (fn [{:keys [sender recipient activity-type]}]
-                  (let [receiving-activity? (= activity-type constants/wallet-activity-type-receive)
-                        relevant-address    (if receiving-activity? recipient sender)]
-                    (= relevant-address current-viewing-account-address))))
-        (distinct)
-        (group-by (fn [{:keys [timestamp]}]
-                    (datetime/timestamp->relative-short-date (* timestamp 1000))))
-        (map (fn [[date activities]]
-               {:title date :data activities :timestamp (:timestamp (first activities))}))
-        (sort-by (fn [{:keys [timestamp]}] (- timestamp))))))
+ :<- [:wallet/network-details]
+ (fn [[activities current-viewing-account-address network-details]]
+   (let [chain-id->network-name (update-vals (group-by :chain-id network-details)
+                                             (comp :network-name first))]
+     (->> current-viewing-account-address
+          (get activities)
+          (keep #(process-activity-by-type chain-id->network-name %))
+          (group-by (fn [{:keys [timestamp]}]
+                      (datetime/timestamp->relative-short-date (* timestamp 1000))))
+          (map (fn [[date activities]]
+                 {:title     date
+                  :data      activities
+                  :timestamp (:timestamp (first activities))}))
+          (sort-by (fn [{:keys [timestamp]}] (- timestamp)))))))
```

Comment:
Added `:wallet/network-details` to properly report the involved network.
Also `process-activity-by-type` is having all the calculations previously made in the view file.